tests/flow/test_graph_info.py

```diff
@@ -83,16 +83,34 @@ def consumeStream(self, stream, drop=True):
 
         self.env.assertEquals(t, "stream")
 
-        # consume entire stream
-        events = self.conn.xrevrange(stream, '+', '-')
-
         # convert stream events to LoggedQueries
-        logged_queries = [LoggedQuery(e[1]) for e in events]
+        logged_queries = []
+        streams = {stream: '0-0'}
+
+        elapsed = 10
+        while len(logged_queries) < n_items and elapsed > 0:
+            # read messages from the stream
+            messages = self.conn.xread(streams, block=0)
+
+            # process each message received
+            stream_messages = messages[0][1]
+            for message_id, message_payload in stream_messages:
+                # logged_queries.insert(0, LoggedQuery(message_payload))
+                logged_queries.append(LoggedQuery(message_payload))
+
+            if stream_messages:
```

Comment:
Do we really need this condition ?

Reply:
We do need such a check, but it should be for `messages` actually. Updated.